src/Shared/Traits.cs

```diff
@@ -94,6 +94,11 @@ public Traits()
         /// </summary>
         public readonly int LogPropertyTracking = ParseIntFromEnvironmentVariableOrDefault("MsBuildLogPropertyTracking", 0); // Default to logging nothing via the property tracker.
 
+        /// <summary>
+        /// Setting this environment variable to 1 disables the node reuse feature.
+        /// </summary>
+        public readonly bool DisableNodeReuse = Environment.GetEnvironmentVariable("MSBUILDDISABLENODEREUSE") == "1";
```

Comment:
nit: Would it make sense to call this from `BuildParameters.Initialize()` (line ~909 in BuildParameters.cs) so there's only one place where we read the environment variable?

Reply:
Good call. I'll go ahead and replace every manual check for this flag with this new property.